belgif-rest-problem-it/belgif-rest-problem-openapi-validation-spring-it/src/main/java/io/github/belgif/rest/problem/OpenApiValidationConfig.java

```diff
@@ -0,0 +1,48 @@
+package io.github.belgif.rest.problem;
+
+import jakarta.servlet.Filter;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import com.atlassian.oai.validator.OpenApiInteractionValidator;
+import com.atlassian.oai.validator.report.LevelResolver;
+import com.atlassian.oai.validator.report.ValidationReport;
+import com.atlassian.oai.validator.springmvc.OpenApiValidationFilter;
+import com.atlassian.oai.validator.springmvc.OpenApiValidationInterceptor;
+
+@Configuration
+public class OpenApiValidationConfig {
+
+    @Bean
+    public Filter validationFilter() {
+        return new OpenApiValidationFilter(true, false);
+    }
+
+    @Bean
+    public WebMvcConfigurer addOpenApiValidationInterceptor(@Value("/openapi.yaml") final String apiSpecification,
+            @Value("server.servlet.context-path") String contextPath) {
+        final OpenApiInteractionValidator validator = OpenApiInteractionValidator
+                .createForSpecificationUrl(apiSpecification)
+                .withLevelResolver(LevelResolver.create()
+                        // Accept unknown query parameters
+                        .withLevel("validation.request.parameter.query.unexpected", ValidationReport.Level.IGNORE)
```

Comment:
I think this check could be decided per-API what the desired behavior would be.
Validation behavior isn't really specified in the OpenAPI standard in case of unknown query parameters.

In case it isn't ignored, I think we should return a badRequest problem with a `urn:problem-type:belgif:input-validation:unknownInput` issue type [[req-valid]](https://www.belgif.be/specification/rest/api-guide/#rule-req-valid).